dapps/profile.vue/src/components/sharings/utils.ts

```diff
@@ -0,0 +1,141 @@
+    const propertiesKeys = Object.keys(properties);
+
+    // check write permissions
+    if (sharing.readWrite) {
+        // check full access
+        if (sharing.readWrite.every(item => propertiesKeys.includes(item))) {
+            return PermissionTypes.FULL_ACCESS;
+        }
+        // check read and write permissions
+        if (sharing.readWrite.some(read => propertiesKeys.includes(read))) {
+            return PermissionTypes.READ_WRITE;
+        }
+    }
+
+    if (sharing.read) {
+        // check full read
+        if (sharing.read && sharing.read.every(item => propertiesKeys.includes(item))) {
+            return PermissionTypes.FULL_READ;
+        }
+        // check read permissions
+        if (sharing.read.some(read => propertiesKeys.includes(read))) {
+            return PermissionTypes.READ;
+        }
+    }
+
+    return PermissionTypes.NONE;
+};
+
+/**
+ * get container from container address
+ * @param runtime
+ * @param containerAddress
+ * @param accountId
+ */
+export const getContainer = (runtime, containerAddress, accountId) => {
+    return new Container(
+        runtime,
+        {
+            accountId,
+            address: containerAddress,
+        },
+    );
+};
+
+/**
+ * get all properties from container
+ * @param Container container
+ */
+export const getContainerProperties = async (container: Container) => {
+    const plugin = await container.toPlugin();
+
+    return plugin.template.properties;
+};
+
+/**
+ * load the contacts for the current user, so we can display correct contact alias
+ * @param runtime current runtime
+ */
+
+export const getContacts = async (runtime): Promise<ContactInterface[]> => {
+    delete runtime.profile.trees[runtime.profile.treeLabels.addressBook]; // why?
+    let addressBook = (await runtime.profile.getAddressBook()).profile;
+
+    return Object.keys(addressBook).map(key => {
+        return {
+            'label': addressBook[key].alias,
+            'value': key
+        };
+    });
+};
+
+
+/**
+ * get permissions from container address
+ * @param runtime
+ * @param containerAddress
+ * @param accountId
+ */
+export const getPermissions = async (runtime, containerAddress, accountId = runtime.activeAccount) => {
+    const container = getContainer(runtime, containerAddress, accountId);
+    const properties = await getContainerProperties(container);
+    const shareConfigs = await container.getContainerShareConfigs();
+    const contacts = await getContacts(runtime);
+
+    let configs =  shareConfigs.map(config => {
+        // the own account should not take into consideration
+        if (config.accountId === accountId) {
+             return null;
+        }
+
+        const contact = contacts.find(item => {
+            return item.value === config.accountId;
+        });
+
+        if (!contacts) {
+            return null;
+        }
+
+        return {
+            accountId: config.accountId,
+            alias: contact.label,
```

Comment:
```suggestion
            alias: contact ? contact.label : config.accountId,
```

adds a fallback if the user has no alias in the address book

![image](https://user-images.githubusercontent.com/763247/67014060-78bf4c00-f0f4-11e9-9441-189a43ca1f38.png)

Reply:
i removed the alias param. Because the ProfilePreview component should handle this!